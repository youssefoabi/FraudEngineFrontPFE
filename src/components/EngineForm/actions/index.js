export const CREATE_ENGINE_REQUEST = 'CREATE_ENGINE_REQUEST';
export const CREATE_ENGINE_SUCCESS = 'CREATE_ENGINE_SUCCESS';
export const CREATE_ENGINE_ERROR = 'CREATE_ENGINE_ERROR';
export const PREVIEW_ENGINE = 'PREVIEW_ENGINE';
export const EDIT_ENGINE_REQUEST = 'EDIT_ENGINE_REQUEST';
export const EDIT_ENGINE_SUCCESS = 'EDIT_ENGINE_SUCCESS';
export const EDIT_ENGINE_ERROR = 'EDIT_ENGINE_ERROR';
export const RESET_ENGINE_SAVE_ERROR = 'RESET_ENGINE_SAVE_ERROR';
export const RESET_ENGINE_SAVE_SUCCESS = 'RESET_ENGINE_SAVE_SUCCESS';
