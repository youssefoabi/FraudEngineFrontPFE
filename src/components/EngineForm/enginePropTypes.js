import { arrayOf, bool, date, number, shape, string } from 'prop-types';

export const scopesPropTypes = arrayOf(
  shape({
    scopeId: number,
    name: string,
    order: number,
    isEnable: bool,
  }),
);

export const parametersPropTypes = arrayOf(
  shape({
    id: number,
    scopeId: number,
    key: string,
    externalCodeId: number,
  }),
);

export const fieldsPropTypes = arrayOf(
  shape({
    id: number,
    type: string,
    order: number,
    label: string,
    isEnable: bool,
    isMandatory: bool,
    parameters: parametersPropTypes,
  }),
);

export const backgroundImagesPropTypes = arrayOf(
  shape({
    order: number,
    urlImageDesktop: string,
    urlLinkDesktop: string,
    urlImageMobile: string,
    urlLinkMobile: string,
    alt: string,
    openInNewTab: true,
    isEnable: true,
  }),
);

export const enginePropTypes = shape({
  id: number,
  code: number,
  name: string,
  isEnable: bool,
  scopes: scopesPropTypes,
  inputFields: fieldsPropTypes,
  backgroundImages: backgroundImagesPropTypes,
  marketingText: string,
  isMarketingTextEnable: bool,
  searchText: string,
  createdAt: date,
  modifiedAt: date,
});
