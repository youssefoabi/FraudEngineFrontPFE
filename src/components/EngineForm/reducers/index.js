export { navigationEngineError, navigationEngineSuccess } from './navigationReducer';
