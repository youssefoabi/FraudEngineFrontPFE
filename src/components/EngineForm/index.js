import { func } from 'prop-types';
import { Grid, Snackbar } from '@material-ui/core';
import { pathOr } from 'ramda';
import { useDispatch, useSelector } from 'react-redux';
import config from 'react-global-configuration';
import MuiAlert from '@material-ui/lab/Alert';
import React, { useCallback } from 'react';

import { enginePropTypes } from './enginePropTypes';
import { RESET_ENGINE_SAVE_ERROR, RESET_ENGINE_SAVE_SUCCESS } from './actions';
import GeneralInformationsSettings from '../GeneralInformationsSettings';
import SearchCriteriaSettings from '../SearchCriteriaSettings';
import VisualSettings from '../VisualSettings';

const EngineForm = ({ engine, setEngine }) => {
  const onDataEditChange = ({ id, value }) => {
    if (id === 'id' && config.get('IsCodeEngineFieldActivated')) return null;

    return setEngine(prevEngine => ({
      ...prevEngine,
      [id]: value,
    }));
  };

  const dispatch = useDispatch();

  const onCloseErrorSnackbar = useCallback(() => dispatch({ type: RESET_ENGINE_SAVE_ERROR }), [dispatch]);
  const onShowSuccessSnackbar = useCallback(() => dispatch({ type: RESET_ENGINE_SAVE_SUCCESS }), [dispatch]);

  const error = useSelector(pathOr(null, ['navigation', 'engineFormError', 'errorMessage']));
  const success = useSelector(pathOr(null, ['navigation', 'engineFormSuccess', 'engineSaveSuccess']));

  return (
    <Grid container direction="column">
      <GeneralInformationsSettings engine={engine} handleChange={onDataEditChange} />
      <VisualSettings engine={engine} handleChange={onDataEditChange} />
      <SearchCriteriaSettings engine={engine} handleChange={onDataEditChange} />
      {error && (
        <Snackbar open autoHideDuration={6000} onClose={onCloseErrorSnackbar}>
          <MuiAlert onClose={onCloseErrorSnackbar} severity="error">
            {error}
          </MuiAlert>
        </Snackbar>
      )}
      {success && (
        <Snackbar open autoHideDuration={6000} onClose={onShowSuccessSnackbar}>
          <MuiAlert onClose={onShowSuccessSnackbar} severity="success">
            Le moteur de recherche a bien été enregistré
          </MuiAlert>
        </Snackbar>
      )}
    </Grid>
  );
};

EngineForm.propTypes = {
  engine: enginePropTypes,
  setEngine: func,
};

EngineForm.defaultProps = {
  engine: null,
  setEngine: Function.prototype,
};

export default EngineForm;
