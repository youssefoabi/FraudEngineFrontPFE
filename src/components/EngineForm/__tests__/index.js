import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import EngineForm from '../';

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  DragDropContext: ({ children }) => children,
}));

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
  useSelector: jest
    .fn()
    .mockReturnValueOnce('ERROR')
    .mockReturnValueOnce(false)
    .mockReturnValueOnce(false)
    .mockReturnValueOnce(true)
    .mockReturnValueOnce(false)
    .mockReturnValueOnce(false),
}));

describe(EngineForm.name, () => {
  describe('with error', () => {
    itRendersCorrectly(<EngineForm />);
  });
  describe('with success', () => {
    itRendersCorrectly(<EngineForm />);
  });
  describe('without Snackbar', () => {
    itRendersCorrectly(<EngineForm />);
  });
});
