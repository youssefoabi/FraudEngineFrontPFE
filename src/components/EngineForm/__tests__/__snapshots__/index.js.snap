// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngineForm with error renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <GeneralInformationsSettings
    engine={null}
    handleChange={[Function]}
  />
  <VisualSettings
    engine={null}
    handleChange={[Function]}
  />
  <SearchCriteriaSettings
    engine={null}
    handleChange={[Function]}
  />
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={6000}
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(Alert))
      onClose={[Function]}
      severity="error"
    >
      ERROR
    </WithStyles(ForwardRef(Alert))>
  </WithStyles(ForwardRef(Snackbar))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`EngineForm with success renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <GeneralInformationsSettings
    engine={null}
    handleChange={[Function]}
  />
  <VisualSettings
    engine={null}
    handleChange={[Function]}
  />
  <SearchCriteriaSettings
    engine={null}
    handleChange={[Function]}
  />
  <WithStyles(ForwardRef(Snackbar))
    autoHideDuration={6000}
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(Alert))
      onClose={[Function]}
      severity="success"
    >
      Le moteur de recherche a bien été enregistré
    </WithStyles(ForwardRef(Alert))>
  </WithStyles(ForwardRef(Snackbar))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`EngineForm without Snackbar renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <GeneralInformationsSettings
    engine={null}
    handleChange={[Function]}
  />
  <VisualSettings
    engine={null}
    handleChange={[Function]}
  />
  <SearchCriteriaSettings
    engine={null}
    handleChange={[Function]}
  />
</WithStyles(ForwardRef(Grid))>
`;
